src/collection.ts

```diff
@@ -636,8 +646,10 @@ export class Collection implements OperationParent {
       throw new TypeError('`options` parameter must not be function');
     }
 
+    const topology = getTopology(this);
+    if (!topology) throw new MongoClientClosedError('Collection.prototype.find');
```

Comment:
Rather than create a custom error message and throwing them in place, it looks like you can actually just move the error inside `getTopology`. `getTopology` should throw if it's unable to find the topology, with a `MongoError` which says that the client has been closed. Additionally, there is no need to include the name of the method in the error message string because you'll get that context in the stack trace anyway.